### DISCUSSION TITLE
Persisting Common Searches

### DISCUSSION THREAD

**Discussion by simonbooth1968** [codefori/vscode-ibmi#2327] [Q&A] [answered]
is there a way of persisting common searches? 

I find that if I do a search, which them populates the treeview, if I then go and do something else and want to come back it's no longer available.

**Comment by worksofliam** [comment]
Are you referring to member/streamfile searches? The list should remain open until:

1. you start a new search, or
2. explicitly close the search results

Are you finding this not to be the case?

**Reply by simonbooth1968** [reply]
Hi Liam,

Thanks for the reply. I am finding that once I get the search results and go some place else, like Explorer, and I return back to the IBM i extension the search is gone. 

It would be nice to have something like a search explorer, ie give it a label and saves it. Many a time have I had to repeat searches.

KR - Simon

**Reply by worksofliam** [reply]
Weird! I typically like to pull the search results over to the over side of the window (usually the right side, called the secondary side bar) and then they will remain until forcefully closed.

**Comment by chrjorgensen** [comment, marked as answer]
@simonbooth1968 When you have your search results open, please note the additional icon in the activity bar:

![billede](https://github.com/user-attachments/assets/71305f63-2ed8-45d7-a7d6-abfd4b2ecae1)

The icon is the same icon as the built-in search function (a bit confusing).

You can click on this icon to get back to your search results.

**Reply by simonbooth1968** [reply]
@chrjorgensen  Thanks for pointing that out. That is exactly what I was after.

**Reply by chrjorgensen** [reply]
@simonbooth1968 Happy to be able to help.
